pkg/odo/cli/dev/dev.go

```diff
@@ -0,0 +1,136 @@
+package dev
+
+import (
+	"fmt"
+	dfutil "github.com/devfile/library/pkg/util"
+	"github.com/redhat-developer/odo/pkg/devfile/adapters/kubernetes"
+	"github.com/redhat-developer/odo/pkg/envinfo"
+	"github.com/redhat-developer/odo/pkg/odo/cli/component"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/spf13/cobra"
+	"io"
+	"k8s.io/kubectl/pkg/util/templates"
+	"os"
+	"path/filepath"
+)
+
+// RecommendedCommandName is the recommended command name
+const RecommendedCommandName = "dev"
+
+type DevOptions struct {
+	// Context
+	*genericclioptions.Context
+
+	// Clients
+	clientset *clientset.Clientset
+
+	// Variables
+	ignorePaths []string
+	out         io.Writer
+}
+
+func NewDevOptions() *DevOptions {
+	return &DevOptions{
+		out: os.Stdout,
+	}
+}
+
+var devExample = templates.Examples(`
+	# Deploy component to the development cluster
+	%[1]s
+`)
+
+func (o *DevOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+func (o *DevOptions) Complete(cmdline cmdline.Cmdline, args []string) error {
+	var err error
+
+	o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline).NeedDevfile(""))
+	if err != nil {
+		return fmt.Errorf("unable to create context: %v", err)
+	}
+	fmt.Fprintf(o.out, "Using devfile.yaml from the current directory.\n")
+
+	envFileInfo, err := envinfo.NewEnvSpecificInfo("")
+	if err != nil {
+		return fmt.Errorf("unable to retrieve configuration information: %v", err)
+	}
+
+	if !envFileInfo.Exists() {
+		// if env.yaml doesn't exist, get component name from the devfile.yaml
+		var cmpName string
+		cmpName, err = component.GatherName(o.EnvSpecificInfo.GetDevfileObj(), o.GetDevfilePath())
+		if err != nil {
+			return fmt.Errorf("unable to retrieve component name: %w", err)
+		}
+		// create env.yaml file with component, project/namespace and application info
+		// TODO - store only namespace into env.yaml, we don't want to track component or application name via env.yaml
+		err = envFileInfo.SetComponentSettings(envinfo.ComponentSettings{Name: cmpName, Project: o.GetProject(), AppName: "app"})
+		if err != nil {
+			return fmt.Errorf("failed to write new env.yaml file: %w", err)
+		}
+	} else if envFileInfo.GetComponentSettings().Project != o.GetProject() {
+		// set namespace if the evn.yaml exists; that's the only piece we care about in env.yaml
+		err = envFileInfo.SetConfiguration("project", o.GetProject())
+		if err != nil {
+			return fmt.Errorf("failed to update project in env.yaml file: %w", err)
```

Comment:
I would like to do that optimization when we have decided what we want to do about "app" construct. Then the operations we perform using `envinfo` package would need to be refactored whether we decide to keep "app" or remove it, because we definitely don't want to be checking/using `envinfo` to figure out the component name.